dataclients/kubernetes/clusterclient.go

```diff
@@ -486,9 +580,23 @@ func (c *clusterClient) fetchClusterState() (*clusterState, error) {
 		return nil, err
 	}
 
-	endpoints, err := c.loadEndpoints()
-	if err != nil {
-		return nil, err
+	endpointSlices, err := c.loadEndpointSlices()
+	// if err != nil {
+	// 	// TODO(sszuecs): ignore because of tests not having endpointslices
+	// 	return nil, err
+	// }
+
+	// TODO(sszuecs) deprecated
+	endpoints, err2 := c.loadEndpoints()
+	// if err != nil {
+	// 	return nil, err
+	// }
+	if endpoints == nil && endpointSlices == nil {
```

Comment:
any idea how to solve this good?
Basically it's enough to fetch one of these (endpointslices or endpoints) if for example permissions are not set correctly or we run in a cluster that has no endpointslices.

Reply:
Lets do endpoint-slices flag. if false fetch and use endpoints otherwise endpointslices

Reply:
done